### PR TITLE
Remove ESLint dependency

### PR DESCRIPTION
**Pull request by davidmrdavid**
Our previous ESLint dependency, which we use for auto-formatting, was being flagged by `npm audit`. We're removing it temporarily until we can an alternative configuration or alternative dependency.

I'm also updating our `mocha` version, for running unit tests.
And simplifying our build-script steps.

This is fairly routine PR, no features should be impacted

### CODE REVIEW

**Review by amdeel — APPROVED** [review]
LGTM!

File: package.json
Comment (ejizba) [inline package.json:70]: Eslint is the standard for Node.js and TypeScript in particular. It does a lot more than just formatting - it's primarily for preventing silly mistakes in code (like not awaiting a promise). I would highly recommend re-enabling eslint ASAP or never removing it in the first place.

Can you quickly try using v7 instead of v6 and see if that works? v7 should get rid of the npm audit complaints and you can turn off any failing rules in [here](https://github.com/Azure/azure-functions-durable-js/blob/dev/.eslintrc) instead of disabling _all_ of eslint. I do think it's fine to get rid of "prettier-eslint", though, because I don't think you're using it. Feel free to ping me if you have issues

Comment (davidmrdavid) [reply package.json:70]: I'll reach out internally on Monday and block this merge for a little bit. Thanks for reaching out!

Comment (ejizba) [reply package.json:70]: No problem, just subscribed to the repo like last week 😁 👀

always watching

Comment (davidmrdavid) [reply package.json:70]: @ejizba: I managed to get this working by upgrading `eslint` and its related dependencies. Could I please get another review? Thanks

**Review by ejizba — APPROVED** [review]

**Pull request merged**
